Feature/plc4py

ottlukas opened the pull request:
This Pull request aims to generate python code using the code generation utils. to create the basis for developing plc drivers in python. Using mspec and code generation

JulianFeinauer committed:
PLC4X-207 When a Handler Timeout occurs cancel the read future to notify the user.

Author:    Julian Feinauer <j.feinauer@pragmaticminds.de>
Signed-off-by: Łukasz Dywicki <luke@code-house.org>

foxpluto committed:
Chached Poll doc

foxpluto committed:
more details

splatch committed:
PLC4X-265 Support for hex mapping of byte/byte[] values in XML test suites.

JulianFeinauer committed:
Started Integration...

JulianFeinauer committed:
Finished Integration.

JulianFeinauer committed:
Finished Integration.

JulianFeinauer committed:
Finished Integration.

JulianFeinauer committed:
Added Copyright Headers.

JulianFeinauer committed:
Renamed to pool2 and readded "old" connection-pool.

JulianFeinauer committed:
Minor fix in pom.

JulianFeinauer committed:
Renamed Pool2 -> Connection-Cache

chrisdutz committed:
- Fixed some commented out code.

ottlukas commented:
https://cwiki.apache.org/confluence/display/PLC4X/Python+-+Built+Utils

ottobackwards commented:
Please add a description of what you are trying to do in this PR and any verification instructions.

ottobackwards commented on pyproject.toml:
```diff
@@ -0,0 +1,14 @@
+[tool.poetry]
+name = "plc4x"
+version = "0.1.0"
+description = ""
```
I'm not sure if this should be individuals or just plc4x project or something.  @chrisdutz ?  if individuals, add me :)

chrisdutz replied:
In general we should never put individuals listed anywhere ... it's part of the legal shield the ASF can protect us with.

ottobackwards replied:
we need to tell the folks at commons that then yikes

chrisdutz committed:
- Implemented the logic to decode properties.

vemmert committed:
PLC4Py maven integration

vemmert committed:
PLC4Py maven integration

vemmert committed:
PLC4Py maven integration

hutcheb committed:
Add future to DisconnectedEvent

When closing a connection, a call to Disconnect has been added
This allows the connection to be closed gracifully.
It will block until the connection is closed, or a timeout occurs before
it closes the channel

hutcheb committed:
Tidied up disconnect logic

Added system property to override option to wait for disconnection to occur.
Added reminder methods for each driver to explicitly override await disconnect option.

splatch committed:
PLC4X-270 Ads driver does not accept double-digit array indexes

chrisdutz commented on pyproject.toml:
```diff
@@ -0,0 +1,14 @@
+name = "plc4x"
+version = "0.1.0"
+description = ""
+authors = ["ottlukas <ott.lukas.14@gmail.com>"]
```
This should better be "Apache PLC4X PMC"

ottlukas replied:
yes ... strange in PyCharm it looks like this:
![image](https://user-images.githubusercontent.com/19743877/104427110-2719a180-5583-11eb-93ae-5dfa0b1b60c4.png)
But when I push to repo it does not...any ideas how to change this (seems to be IDE related)

chrisdutz commented on sandbox/plc4py/pyproject.toml:
```diff
@@ -24,6 +24,7 @@ authors = ["Apache PLC4X <dev@plc4x.apache.org>"]
+numpy = "^1.19"
 
 [tool.poetry.dev-dependencies]
 pytest = "^5.2"
```
Wouldn't it be better to put explicit versions in here and not version ranges? (Or is this more like our prerequisite check in PLC4X where we check at least a given version is installed?)

ottlukas replied:
1) Already deleted numpy
2) We will follow PEP-517 https://www.python.org/dev/peps/pep-0517/ and define which minimum version of a module is required for the building.

chrisdutz replied:
But doesn't this make things not reproducible?

hutcheb committed:
Added some logging

hutcheb committed:
Add confluent package to deployment so it appears in Nexus.

https://repository.apache.org/content/repositories/snapshots/org/apache/plc4x/plc4j-apache-kafka/

chrisdutz committed:
- Added the missing drivers

chrisdutz committed:
- Try to fix the deployment issue.

hutcheb committed:
Updated Notice and License files.

To take into account for derivative works licenses under EPL 2.0

chrisdutz committed:
- Added the extensions to the LICENSE and NOTICE files which maven will generate in the Kafka connector archives

chrisdutz committed:
- Minor updates (Generally the order of imports so a reformat doesn't change the files too much)

chrisdutz committed:
- Minor updates (Generally the order of imports so a reformat doesn't change the files too much)

hutcheb committed:
Add Encryption handler for opcua server

chrisdutz committed:
- Continuing work on the C S7 driver (Curenlty the changes in mspec are not correct ... need to continue there)

chrisdutz committed:
PLC4X-272
When splitting up large requests, too big sub-requests are generated (S7)
- When splitting up into more than one request, the new request was correctly initialized with the empty-request size, however the current element wasn't added. This should now be fixed

chrisdutz committed:
PLC4X-272
When splitting up large requests, too big sub-requests are generated (S7)
- When splitting up into more than one request, the new request was correctly initialized with the empty-request size, however the current element wasn't added. This should now be fixed

chrisdutz committed:
PLC4X-272
When splitting up large requests, too big sub-requests are generated (S7)
- When splitting up into more than one request, the new request was correctly initialized with the empty-request size, however the current element wasn't added. This should now be fixed

hutcheb committed:
Re-Wrote OPC UA server -> Milo Interface

Outstanding issue with PLC4X comms failure when adding tag that isn't valid.

hutcheb committed:
Finialized OPCUA server and updated Kafka connector

Fixed opcua server issue when polling errors occur. Now polling doesn't
stop for other tags.

Updated kafka conector based on Rankesh's comment about adding jitter to
the poll return time.

hutcheb committed:
Updated based on Ottos and Matthias' comments.

Rename Encryption file to CertificateGenerator
Updated license and notice files.

Fix, if certificate store doesn't exist a new one wasn't created.

chrisdutz committed:
- Made the Reader detect the case of a property not existing or not having the permission to read.

chrisdutz committed:
- Fixed some package errors

chrisdutz committed:
- Adjusted the "TestKnxNetIpPlc4goDiscovery" test to work with the updated Browse functionality.

chrisdutz committed:
- Made the Udp and Tcp transport not log an error in GetNumReadableBytes

chrisdutz committed:
- Removed an unneeded log output

chrisdutz committed:
- Made the KNX driver return an error if the device we try to connect to doesn't support "tunneling"

chrisdutz committed:
- Added more documentation on KNX operations

chrisdutz committed:
- Added stuff for decoding the KNX DeviceDescriptors

chrisdutz committed:
- Fixed a wrong package name

chrisdutz committed:
- Replaced the manually created KnxDatatypes and DataIo with a fully generated version

chrisdutz committed:
- Updated generated Go code

chrisdutz committed:
- Finished migrating the Go KNX client to the new model.

chrisdutz committed:
- Some version updates of dependencies prior to cutting the release branch

chrisdutz committed:
- Updated the list of fixed bugs in the RELEASE_NOTES

chrisdutz committed:
- Added the new programming languages to doe doap file

chrisdutz committed:
- Ensured all integrations and examples have all of our drivers available
- Removed the "with-proxies" profile and everything that relies on this
- Updated the prerequisiteCheck.groovy script to no longer check thrift-related things
- Updated the release and release-validation sites

chrisdutz committed:
- Made sure the generated code for accessing properties of a variable are start upercase

hutcheb committed:
removed http port

chrisdutz committed:
[maven-release-plugin] prepare branch @{releaseLabel}

chrisdutz committed:
- Updated the release documentation a bit more

chrisdutz committed:
[maven-release-plugin] prepare branch @{releaseLabel}

chrisdutz committed:
[maven-release-plugin] prepare for next development iteration

chrisdutz committed:
- Cleaned up after preparing the release-branch

chrisdutz committed:
- Cleaned up after preparing the release-branch

chrisdutz committed:
- Cleaned up after preparing the release-branch

chrisdutz committed:
- Cleanup after preparing the release

chrisdutz committed:
- Cleanup after preparing the release

chrisdutz committed:
- Cleanup after preparing the release

chrisdutz committed:
- Made the download_staged_release.sh download a README and not a README.md

chrisdutz committed:
- Fixed the profile names in the Dockerfile (Still doesn't work for me)

chrisdutz committed:
- Merged some changes from the release branch

chrisdutz committed:
- Fixed the Docker build on Windows systems

chrisdutz committed:
- Added dataIo of the KNX property types
- Refactored the Go KNX driver to support the concept of device-connections

chrisdutz committed:
- Some minor bug-fixing of my latest changes

chrisdutz committed:
- Updated the download and doap files after officially releasing 0.8.0

chrisdutz committed:
- Switched from printlines to logging
- Made sure timeouts are applied everywhere where we have blocking operations
- Made sure new go routines are started where needed in order to prevent lockups

chrisdutz committed:
- Added an option to directly intercept on found resources (Allowing immediate interaction)

chrisdutz committed:
- Refactored the callback-based operations

hutcheb committed:
Update manifest based on Rankesh's request.

chrisdutz committed:
PLC4X-276 - [S7] The most of the supported types don't work correctly
- Fixed this by extending the syntax for enum fields to allow providing a "field name" to use for serialization/deserialization

NO JIRA:
- Continued working on getting more of the S7 datatypes working in the PLC4C S7 driver (All except the temporal types now work)

chrisdutz committed:
- Finished the missing temporal types

chrisdutz committed:
- Fixed the RELEASE_NOTES

chrisdutz committed:
- Fixed the RELEASE_NOTES

chrisdutz committed:
- Added Social-Media Release image template

chrisdutz committed:
- Added Apache Header to release-image template

chrisdutz committed:
- Removed obsolete type

chrisdutz committed:
- Removed more obsolete types

foxpluto committed:
fix logback dependency

foxpluto committed:
moved to 0.9.0 version. The code is stable

chrisdutz committed:
- Managed the dependencies

chrisdutz committed:
- Managed a dependency which was causing trouble
- Renamed the connection cache module to output "Connection Cache" and not "Connection Pool" in the build summary

chrisdutz committed:
- Updated Antlr4 version

chrisdutz committed:
- Implemented the MemRead functionality for KNX
- Implemented a demo to read a knx devices group-address table

chrisdutz committed:
- Implemented the reading and decoding of the group-addres association table.

foxpluto committed:
new documentation for Connection Cache

splatch committed:
PLC4X-280 Fix position of fields in ADS discovery.

foxpluto committed:
connection cache doc page for the site

chrisdutz committed:
- Refactoring of the KNX LData messages
- Finished a first working KNX Group-Address & ComObject discovery

chrisdutz committed:
- Made the driver more robust against not 100% compliant devices

chrisdutz committed:
- Added a missing service type "Routing"

chrisdutz committed:
- Refactored the callback-based operations

chrisdutz committed:
- Enabled a full KNX network scan instead of one limited to my address-ranges

ottlukas committed:
hyphens

https://issues.apache.org/jira/browse/PLC4X-282 
"No license-costs" should not be hyphenated on the main index page

ottlukas committed:
updating webpage

https://issues.apache.org/jira/browse/PLC4X-282

https://issues.apache.org/jira/browse/PLC4X-281

ottlukas committed:
fix download link 0.7

chrisdutz committed:
- Removed the length-check for the memory-read response handling

chrisdutz committed:
- Started cleaining up before starting to work on the Modbus driver

chrisdutz committed:
- Implemented the field address parser for modbus addresses

chrisdutz committed:
- Global refactoring of the way Modbus DataTypes are handled
- Continued implementing the C Modbus driver

chrisdutz committed:
- Added a 10ms sleep in case of not having any incoming data for modbus as otherwise the receive thread was going nuts.

chrisdutz committed:
- Make the read-memory functionality more robust against misbehaving devices

chrisdutz committed:
- Fixed a bug not processing group-addresses, if only one item is returned

chrisdutz committed:
- Add code to auto acknowledge unacknowledged knx messages (Hopefully preventing connection problems)

chrisdutz committed:
- Bumped the severity level of some log messages to simplify tracking down the issues.

chrisdutz committed:
- Added some logging to the timer for the keep-alive to see if it's started multiple times.

chrisdutz committed:
- Disabled the keep alive timer completely

chrisdutz committed:
- Fixed the way errors are reported (put the pushing of the result into the result pipe into a go-routine)

chrisdutz committed:
- Adjusted the write example the same way the read-example was updated

chrisdutz committed:
- Finished implmenting the processing of Modbus read-responses in the PLC4C Modbus Driver

chrisdutz committed:
- Implemented active Group Address reads
- Greatly refactored how the group-address reading works

chrisdutz committed:
- Introduced a "BlockingClose" for usage with defer statements

chrisdutz committed:
- Introduced a "BlockingClose" for usage with defer statements

chrisdutz committed:
- Fixed the output of the example

chrisdutz committed:
- Removed an unneeded log output.

hutcheb committed:
Add support for variable strings and enum discriminators.

chrisdutz committed:
- Updated examples to register the drivers in a way that works outside the project.

hutcheb committed:
Integration test works with existing test protocol.

Need to remove my test protocol I added
Add Integration test to other protocols
Fill out the test protocol with valid mspec that covers all documented types.

hutcheb committed:
Cleaned up build-utils integration tests.

Still have an issue with building the c and go modules
For C it relies on the spi module but hasn't been compiled yet.

chrisdutz committed:
- Implemented Device Authentication
- Implemented reading of Property Descriptions
- Started implementing the code for reading of Com Object Tables

chrisdutz committed:
- Finetuned the go code generation.

chrisdutz committed:
- Fixed a nasty bug in the sending of messages (related to an overflow in the counters)
- Finished implementing the decoding of the Com Object Table for System B devices

hutcheb committed:
Compiled the test protocol mspec in C

Checksum support is missing
Updated the int and uint type selection so they match Java.
there was an issue when selecting non standard sized types.

chrisdutz committed:
- Implemented a tool to fetch all available KNX definitions from the KNX Foundations Servers
- Added definitions for the System7 type com-object-table data
- Refactored the API for Browse in PLC4Go
- Refactored the API for Read/Write/Subscribe to allow adding of "field" object and renamed "addItem" to "addQuery"

chrisdutz committed:
- Finished Go KNX Auto-discovery

chrisdutz committed:
- Removed a duplicated type

hutcheb committed:
Updated connection cache pom based on Cesar's comments.

chrisdutz committed:
- Fixed a minor bug in the knx master-data mspec generation.

chrisdutz committed:
- Fixed a bug only returning results for only one group address for a given com object in System7 devices

chrisdutz committed:
- Added an example for subscriptions in KNX
- Fixed the code in KNX subscriptions to work if no datatype was provided

chrisdutz committed:
- Made the driver replace the "field" in case of a subscription event to replace the original field, which can contain patterns, with a concrete version with no patterns. So we can know where an event came from.

hutcheb committed:
Got enumerator discriminators working for for C

chrisdutz committed:
- Added a "RawPlcValue" type which allows lazy decoding of the payload if the type-information is not known at the time it's processed by the driver.

hutcheb committed:
Updated some wording on the Kafka connector page

chrisdutz committed:
- Added a RawReset function to the RawPlcValue to allow multiple parser runs of the same raw value.

chrisdutz committed:
- Changed the PlcValueAdapter.go to support the list functions on all plc value types, by treating single item plc-values as one-element lists.

hutcheb committed:
Fix for opcua subscription

hutcheb committed:
Updated C and Go templates

Still have to confirm Go generated sources ae being compiled to check for some errors.
Have to rework the java templates as the mspec format changed when implementing C

chrisdutz committed:
- Major cleanup of the KnxNetIpConnection type
- Made the drivers automatically register their default transports

chrisdutz committed:
- Updated the knx codec to check expectations for expirations, even if no message was received
- Added a dedicated TimeoutError type to allow easy detection of timeout errors

chrisdutz committed:
- Clean up the read-memory function and the updating of the max pdu size, if a mem-read returns less than expected.

chrisdutz committed:
- Clean up the read-memory function and the updating of the max pdu size, if a mem-read returns less than expected.

chrisdutz committed:
- Worked on detecting bad connections.

chrisdutz committed:
- Fixed the Firmata driver (The analog part now works again)

chrisdutz committed:
- Removed unneeded import.

chrisdutz committed:
- Removed unneeded import.

chrisdutz committed:
- Tried to resolve a strange problem only causing problems when compiling using gccgo

ottlukas committed:
lebbing logo

ottlukas committed:
adding Lebbing Automation

sruehl committed:
golang type generator big.Int and big.Float support + initial ads generation

sruehl committed:
remove debug statements from golang readbuffer

chrisdutz committed:
- Added an execution of go fmt in the build
- Fixed a bug in the error handling of generated XML code

sruehl committed:
plc4go: add Makefile to wrap maven calls for non maven aware devs

sruehl committed:
plc4go: enabled test run with maven and reformatted missing code

sruehl committed:
plc4j: removed L from spec literal and added detection for long literals to java gernerator

sruehl committed:
plc4j: removed L from spec literal and added detection for long literals to java gernerator Part2 now with spec

sruehl committed:
plc4go: add WIP test files for ads and s7

chrisdutz committed:
- Adjusted the execution times of the go maven plugin tools
- Removed obsolete test-code (That actually wasn't test code)

sruehl committed:
plc4go: severall fixes regarding code generation
+ math/big import will only be rendered when required
+ generated code will now contain a comment which will be interpreted by the IDE
+ reformated code with tabs again. We need to adjust code generation here somehow
+ added prober .editorconfig for .go

chrisdutz committed:
- Fixed the code generation issues for the new protocols

sruehl committed:
plc4go: added parameter parsing to DriverTestRunner.go

sruehl committed:
plc4go: initial work on ads driver
+ introduced zerolog github.com/rs/zerolog
+ introduced errors github.com/pkg/errors
+ Added some logging to DriverTestRunner
+ Fixed bug where delay lead to panic

sruehl committed:
plc4go: Switch to zerolog
+ Fixed issues with tabs by adding tabs to golang ftlh templates
+ Fixed issue where generated header was missing
+ Fixed issue where fmt wasn proberly executed during maven cycle
+ added indent_style for ftlh

sruehl committed:
plc4go: Enabled xunit report xml for jenkins

sruehl committed:
plc4go: Disable console logger as it wrecks the report.xml

chrisdutz committed:
- Tried to make the KNX protocol always update the xml-masterdata.xml

sruehl committed:
plc4go: Make build on test fail

sruehl committed:
plc4go: Only fail test after report was generated

sruehl committed:
plc4go: Partial test fix
+ un/marshal uses hex base output now
+ DriverTestsuite.xml still contains base64

sruehl committed:
plc4x: convert missing base64 to hex

sruehl committed:
plc4j: added deserializer to TestSuiteMappingModule.java

chrisdutz committed:
- Removed the setting to re-download the xml-masterdata.xml on every build as in case of multiple builds a day you tend to get banned.

chrisdutz committed:
- Updated the ads and can protocol modules to use hex-string encoded test-data

chrisdutz committed:
- Updated the s7 and can protocol modules to use hex-string encoded test-data

chrisdutz committed:
- Updated the testsuite data in the go module

chrisdutz committed:
- Deleted some super-old legacy test

hutcheb committed:
Updated Enum Discriinator fields for C and Java.

splatch committed:
Fix legacy transport name and unsubscribe request handling for CANopen.

hutcheb committed:
Integration test is working in Go, sorting through some failed tests.

thomas169 committed:
added strtok_s define for windows

hutcheb committed:
Minor changes

hutcheb committed:
Change default Duration fo change of state to 0 seconds from 1 second.

hutcheb committed:
Removed timestamp from write request.

OPCUA servers shouldn't complain if it isn't there.

chrisdutz committed:
- Updated the website-generation to use the updated skin with the new tooling for Apache events.

chrisdutz committed:
- Updated the commercial-support.adoc page (Replaced codecentric entry with c-ware (as I'm no longer affiliated with cc))

chrisdutz committed:
- Fixed my email address

hutcheb committed:
Integration tests working for java,C and Go

hutcheb committed:
Add additional tests to the test.mspec

sruehl committed:
plc4go: Cleanup obsolete interface child
Implementation meant a implements but was in fact a delegate attribute. So those got removed

sruehl committed:
plc4go: Refactor, deduplicate package name out of code

sruehl committed:
plc4go: Removed obsolete interface delegates
+ added a readable test goal to Makefile
+ added Todos for DriverTestRunner.go

sruehl committed:
plc4go: Improve error output on test failure
+ Provide a diff view for cases when the test fails

sruehl committed:
plc4go: Change logger to text (only for unit tests)
+ Disable color when run from jenkins as the color instructions damage the xunit output

sruehl committed:
plc4go: Cleanup
+ Added knxnetip_driver_test.go and knxnetip_parser_serializer_test.go for intial skipped test runs
+ Deduplicated Package names out of struct names
+ Converted consts to camel case
+ Added timeout for timeout error
+ removed package shadowing on modbus/Reader.go
+ fixed several linter issues

sruehl committed:
plc4go: Fix missing method that broke the build

sruehl committed:
plc4go: (Partial)Fix ordering issues when using maps
- Skipped "Multi element read request" and "Array element read request" tests
+ Testsuite runner reports now sub-tests as proper tests to the golang testing framework
+ fixed issue where Read ran into a deadlock too soon due to writing in a not returned channel
+ Added MultiError.go to collect errors
+ Introduced name slices which keep the order of fields (and queries)
+ refined error messaging

sruehl committed:
plc4go: Skip modbus "Single element read request" test case
All three of these tests seem to have timing issues were they sometimes run sometimes not

sruehl committed:
plc4go: Improved ParserSerializerTestRunner.go with sub-test and skip
+ Testcases are now properly reported as sub tests
+ Added ability to skip test cases

sruehl committed:
plc4go: fix build by adding a header

sruehl committed:
plc4go: refined logging
+ Added a convenience import to suppress too verbose logging for plc4go user which can be used like that `import _ "github.com/apache/plc4x/plc4go/pkg/plc4go/logging"`
+ Added log statements to driverManager.go

sruehl committed:
plc4go: fix concurrency issue in testframework

chrisdutz committed:
- Removed the "(incubating)" from IoTDB

sruehl committed:
plc4go: added more log statements
+ use wrapped errors for models
+ use code generation for const names
+ added several todos and minor fixes

sruehl committed:
plc4go: reactivated test by inserting artificial delay

sruehl committed:
plc4j: added skip support for DriverTestsuiteRunner and ParserSerializerTestsuiteRunner

chrisdutz committed:
- Changed the usage of the download-plugin to a groovy script that ensured the knx-master-data.xml is updated regularly without anoying the KNX foundation server

chrisdutz committed:
- Disabled the string test in the ReadBuffer tests for plc4c as this seems to be failing on windows and needs investigating.

hutcheb committed:
Commmit before refactoring

hutcheb committed:
Refactored the String length expression to a Term.

Cleaned up a little bit.

C and .NET still left to do

hutcheb committed:
Updated the C branch with the String length expression.

updated pom.xml to inherit stuff for the integration tests
Updated the RAT and gitignore to only ignore C Makefiles as these
are automatically generated from cmake.

hutcheb committed:
Fixed various formatting issues.

hutcheb committed:
Removed the local repository requirements for the build-util modules.

sruehl committed:
plc4go: first ads driver draft
+ contains additional minor tweaks

sruehl committed:
plc4go: work on ads driver
+ fixes on model-template.ftlh to enable complex types
+ fixes on enum-template.ftlh to allow usage of constants
+ DriverTestRunner.go now properly appends options
+ fixed possible deadlock in modbus driver
+ fixed possible deadlock in knx driver

sruehl committed:
plc4go: ads works now with a simple read
+ Fixed MultiError as it returned wrong type (TIL (*utils.MultiError = nil) != nil)
+ Fixed proper error handling on DriverTestRunner.go

sruehl committed:
plc4go: use datatype length as read length

sruehl committed:
plc4go: fix issue with unmarshalling on nested lists

chrisdutz committed:
- Fixed the handling of DATE_AND_TIME variants of ADS in Go

sruehl committed:
plc4go: ads first draft of symbolic addressing implemented
+ Test currently doesn't run due to issues on xml (un)marshalling

sruehl committed:
plc4go: fixed issue when unmarshaling list elements from xml

sruehl committed:
plc4go: ads symbolic resolving works (just in the wrong mode)
+ currently works in big endian opposed to the required little endian. Once this is resolved the ignore list entry for "Single element symbolic read request" can be removed

chrisdutz committed:
- Made the integration test use the matching version of CMake for the current operating system (Mac, Linux, Windows)

chrisdutz committed:
- Made sure the required artifacts are built first

sruehl committed:
plc4x: prefix the suffix of language templates with their language suffix to better control editor config

sruehl committed:
plc4x: fix .editorconfig for go templates

sruehl committed:
plc4x: fix .editorconfig for java templates

sruehl committed:
plc4go: ads symbolic resolving works now
+ Added little endian support (basic) in plc4go

sruehl committed:
plc4go: fixes on ads symbolic resolving

+ Enabled test "Single element symbolic read request (Address previously resolved)"
+ Writer.go now uses Reader.go for symbolic resolving
+ Fixed DriverTestRunner.go no freeing channels after use

sruehl committed:
plc4go: partial multi read implementation

sruehl committed:
plc4go: presort imports to avoid unnecessary diffs

sruehl committed:
plc4go: implemented Stringer on models
+ models do have now a pretty String()
+ added a hex dumper to dump binary data as hex+string view
+ added a ascii boxer which renders data in ascii boxes

sruehl committed:
plc4go: fixed Stringer not properly render childs on discriminators

+ if box is called wit "" as name the type will be used

sruehl committed:
plc4go: Stringer now properly renders arrays

+ introduced new BoxBox function to avoid casting the AsciiBox to string
+ added simple rendering of strings as is
+ added inactive code for rendering hex values later as well

sruehl committed:
plc4go: Ported ManualTestRunner.go from java

+ This allows to run a manual test against a real plc

sruehl committed:
plc4go: ManualTestRunner.go now requires the user to construct the driver manager

+ This is due to a limitation of golang

sruehl committed:
plc4go: fix build

+ accidentally commited a manual test (Reminder: find a way to enable them only if explicitly selected)

chrisdutz committed:
- Make Maven tests skippable in Go

sruehl committed:
plc4go: fixed xml marshalling and unmarshalling on arrays

sruehl committed:
plc4go: let template emit required imports

+ Letting the template emitting the required imports simplifies the import calculation as we only rely on a sorted set.

sruehl committed:
plc4go: revert xml element iteration to work only for complex types

sruehl committed:
plc4go: enable last test for ads_driver_test.go

+ Added workaround for AdsReadWriteResponse in case there is only 1 element there is no reading of the response code. This nees further investigation

sruehl committed:
plc4go: fix potential issue with occasional test failures

+ by using the list of fields from the request we avoid the order problems with golang maps witch leads to occasional test failures as the order can be random

sruehl committed:
plc4x: 5c051470 fixed issues in protocol-test for plc4go

+ protocol-test has now todos for open issues

sruehl commented:
@ottlukas maybe it might be a good idea to just rebase this branch onto develop as its really hard to see whats contained within this branch. Should be fairly easy with the interactive-rebase feature of intellij.
Usually you don't rebase pushed branches but as this branch is contained within you personal fork we should be good.
WDYT?

ottlukas commented:
> @ottlukas maybe it might be a good idea to just rebase this branch onto develop as its really hard to see whats contained within this branch. Should be fairly easy with the interactive-rebase feature of intellij.
> Usually you don't rebase pushed branches but as this branch is contained within you personal fork we should be good.
> WDYT?

Yes just rebase is fine with me

ottlukas committed:
Merge remote-tracking branch 'origin/feature/plc4py' into feature/plc4py

# Conflicts:
#	build-utils/language-c/src/main/resources/templates/c/data-io-template.c.ftlh
#	build-utils/language-c/src/main/resources/templates/c/pojo-template.c.ftlh
#	build-utils/language-go/src/main/resources/templates/go/data-io-template.go.ftlh
#	build-utils/language-go/src/main/resources/templates/go/model-template.go.ftlh
#	build-utils/language-java/src/main/java/org/apache/plc4x/language/java/JavaLanguageTemplateHelper.java
#	build-utils/language-java/src/main/resources/templates/java/pojo-template.java.ftlh
#	build-utils/language-python/src/main/java/org/apache/plc4x/language/python/PythonLanguageTemplateHelper.java
#	build-utils/pom.xml
#	plc4go/internal/plc4go/ads/readwrite/model/AdsData.go
#	plc4go/internal/plc4go/ads/readwrite/model/AdsDataType.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetConfirmedServiceRequestConfirmedCOVNotification.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetConfirmedServiceRequestReadProperty.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetConfirmedServiceRequestSubscribeCOV.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetConfirmedServiceRequestWriteProperty.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetErrorReadProperty.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetServiceAckReadProperty.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetTagWithContent.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetUnconfirmedServiceRequestIAm.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetUnconfirmedServiceRequestUnconfirmedPrivateTransfer.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetUnconfirmedServiceRequestWhoHas.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BACnetUnconfirmedServiceRequestWhoIs.go
#	plc4go/internal/plc4go/bacnetip/readwrite/model/BVLC.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduControl.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduControlAck.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduControlConnect.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduControlContainer.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduControlDisconnect.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduControlNack.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduData.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataAdcRead.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataAdcResponse.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataContainer.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataDeviceDescriptorRead.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataDeviceDescriptorResponse.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExt.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtAuthorizeRequest.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtAuthorizeResponse.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtDomainAddressRead.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtDomainAddressResponse.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtDomainAddressSelectiveRead.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtDomainAddressSerialNumberRead.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtDomainAddressSerialNumberResponse.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtDomainAddressSerialNumberWrite.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtDomainAddressWrite.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtFileStreamInfoReport.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtGroupPropertyValueInfoReport.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtGroupPropertyValueRead.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtGroupPropertyValueResponse.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtGroupPropertyValueWrite.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtIndividualAddressSerialNumberRead.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtIndividualAddressSerialNumberResponse.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtIndividualAddressSerialNumberWrite.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/ApduDataExtKeyResponse.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/CEMIAdditionalInformationBusmonitorInfo.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/CEMIAdditionalInformationRelativeTimestamp.go
#	plc4go/internal/plc4go/knxnetip/readwrite/model/KnxNetIpMessage.go
#	plc4go/internal/plc4go/modbus/readwrite/model/ModbusConstants.go
#	plc4go/internal/plc4go/modbus/readwrite/model/ModbusTcpADU.go
#	plc4go/internal/plc4go/s7/readwrite/model/S7Message.go
#	plc4go/internal/plc4go/s7/readwrite/model/S7Payload.go
#	plc4go/internal/plc4go/s7/readwrite/model/S7PayloadUserDataItemCpuFunctionReadSzlResponse.go
#	plc4go/internal/plc4go/s7/readwrite/model/TPKTPacket.go
#	plc4j/drivers/modbus/src/main/java/org/apache/plc4x/java/modbus/protocol/ModbusProtocolLogic.java
#	plc4j/drivers/simulated/src/main/java/org/apache/plc4x/java/simulated/connection/SimulatedDevice.java
#	plc4j/drivers/simulated/src/main/java/org/apache/plc4x/java/simulated/field/SimulatedFieldHandler.java
#	plc4j/integrations/opcua-server/pom.xml
#	plc4j/spi/src/main/java/org/apache/plc4x/java/spi/Plc4xNettyWrapper.java
#	pom.xml
#	protocols/ads/src/main/resources/protocols/ads/ads.mspec
#	protocols/knxnetip/src/main/xslt/knx-types.xsl
#	sandbox/plc4py/pom.xml
#	sandbox/test-java-df1-driver/pom.xml

sruehl commented:
I merge this now into a temporary branch and try to cleanup the history there,

sruehl committed:
Merge branch 'feature/plc4py_temp' into feature/plc4py
